Bobiquette

aflibottesavoie opened the pull request:
Ajout à la classe IA :
- ajout d'un enum direction
- ajout d'explication en commentaire de cet enum
- identification de ce que l'on considère comme i et ce que l'on considère comme j en commentaire en haut
- ajout du calcul des pièces sur une ligne d'actions donnée
- ajout de commentaires dans toutes les méthodes
- correction du déplacement d'une pièce (j'avais inverser I et J)
- ajout d'une méthode pour notify le IA d'un move adverse
- Toutes les valeurs ajouté au tableau solving sont en valeur finale statique dans le haut de la classe (pour des modifs rapide)
- ajout de la méthode letterToIndex
- ajoute de "enemyDevalue" qui identifie sur le solving board que c'est mauvais de manger des pions adverse :D

aflibottesavoie committed:
Version du IA comprenant :
- un petit affichge du board et du solving board
- trois constructeur qui consrtuit automatiquement ce qui manque
- un board de jeu (playBoard)
- un board pour solver (solvingBoard)
- une liste de nos pièces
- une ébauche de calcul des valeurs d'une tuile donnée
- plusieurs méthode pour supporté ce calcul
- une méthode pour faire l'update des positions lorsqu'un move est fait

aflibottesavoie committed:
Ajout à la classe IA :

- ajout d'un enum direction
- ajout d'explication en commentaire de cet enum
- identification de ce que l'on considère comme i et ce que l'on considère comme j en commentaire en haut
- ajout du calcul des pièces sur une ligne d'actions donnée
- ajout de commentaires dans toutes les méthodes
- correction du déplacement d'une pièce (j'avais inverser I et J)
- ajout d'une méthode pour notify le IA d'un move adverse
- Toutes les valeurs ajouté au tableau solving sont en valeur finale statique dans le haut de la classe (pour des modifs rapide)
- ajout de la méthode letterToIndex
- ajoute de "enemyDevalue" qui identifie sur le solving board que c'est mauvais de manger des pions adverse :D

aflibottesavoie committed:
Ajout à la classe IA :

- ajout d'un enum direction
- ajout d'explication en commentaire de cet enum
- identification de ce que l'on considère comme i et ce que l'on considère comme j en commentaire en haut
- ajout du calcul des pièces sur une ligne d'actions donnée
- ajout de commentaires dans toutes les méthodes
- correction du déplacement d'une pièce (j'avais inverser I et J)
- ajout d'une méthode pour notify le IA d'un move adverse
- Toutes les valeurs ajouté au tableau solving sont en valeur finale statique dans le haut de la classe (pour des modifs rapide)
- ajout de la méthode letterToIndex
- ajoute de "enemyDevalue" qui identifie sur le solving board que c'est mauvais de manger des pions adverse :D

brunopc-net commented:
La classe getIndexFromLetter devrait être dans InOut je crois. la classe InOut pourrait faire un genre de IA.getNextMove(int[][] board) -board passé en paramètre- et le IA nous retourne le prochain move, mais c'est bien la classe InOut qui gère les paramètres (coups) passés par le serveur

aflibottesavoie commented:
j'ai bugger lol, ta raison pour le 10x10

pi pour la méthode sa me dérange pas, mais sa doit se rendre a l'algo d'une facon ou d'une autre

aflibottesavoie commented:
j'ai corriger 2-3 choses aussi, je vais en faire un autre bientot.

aflibottesavoie commented:
Deplus, je crois que plutot que d'avoir ia.getnextmove, sa serait plutot de faire rouler le IA sur le plus de tours possibles dans un thread de 4.5 secondes, et l'algo garderais le move le plus intelligent en mémoire

brunopc-net commented:
Moi aussi, une nouvelle version de serveur et InOut bientôt

aflibottesavoie committed:
Ajout de centre de masse
correction des petit glitch

aflibottesavoie committed:
Modification sur la class IA :
- correction des petit glitch (directions)

- correction du centre de masse et de son utilisation

- ajout de pointage sur le solving board dans le cas ou la tuile se trouve entre le pion adverse et son centre de masse

- méthode getCoordAfterMovement qui nous donne les nouvelles coordonées d'une pièce si on fait un déplacement selon un axe

- nos tuiles dans le solving board sont mtn des -10, plutot que des -1 puisque les pions adverse ont aussi des valeurs négative !

- changements de la méthode d'affichage :
        -ajout du centre de masse des deux joueurs à l'affichage
        -changement de nos tuiles pour des X dans le solving board

aflibottesavoie committed:
Modification sur la class IA :
- correction des petit glitch (directions)

- correction du centre de masse et de son utilisation

- ajout de pointage sur le solving board dans le cas ou la tuile se trouve entre le pion adverse et son centre de masse

- méthode getCoordAfterMovement qui nous donne les nouvelles coordonées d'une pièce si on fait un déplacement selon un axe

- nos tuiles dans le solving board sont mtn des -10, plutot que des -1 puisque les pions adverse ont aussi des valeurs négative !

- changements de la méthode d'affichage :
        -ajout du centre de masse des deux joueurs à l'affichage
        -changement de nos tuiles pour des X dans le solving board
